spigot_v1_15_R2/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/Spigot_v1_15_R2.java

```diff
@@ -318,6 +320,42 @@ public BaseBlock getBlock(Location location) {
         return state.toBaseBlock();
     }
 
+    private void updateNeighbours(World world, BlockPosition blockposition, Chunk chunk, IBlockData oldBlock, IBlockData newBlock, IBlockData actualBlock, int i) {
+        if (actualBlock == newBlock) {
+            if (oldBlock != actualBlock) {
+                world.b(blockposition, oldBlock, actualBlock);
+            }
+
+            if ((i & 2) != 0 && (!world.isClientSide || (i & 4) == 0) && (world.isClientSide || chunk == null || chunk.getState() != null && chunk.getState().isAtLeast(
+                    PlayerChunk.State.TICKING))) {
+                ((ChunkProviderServer) world.getChunkProvider()).flagDirty(blockposition);
+            }
+
+            if (!world.isClientSide && (i & 1) != 0) {
+                world.getType(blockposition.west()).doPhysics(world, blockposition.west(), oldBlock.getBlock(), blockposition, false);
```

Comment:
I feel like we can do a loop here to reduce copy/paste:
```java
for (BlockPosition b : ImmutableList.of(bp.east(), bp.west(), ...) {
  world.getType(b).doPhysics(world, b, oldBlock.getBlock(), bp, false);
}
```

There might even already be something to iterate all directions?

Reply:
Yeah, this was done as I copy/pasted it from MC code. I'll do a big cleanup when we decide what to keep etc